### DISCUSSION TITLE
Future ideas

### DISCUSSION THREAD

**Discussion by andymac4182** [markdown-confluence/markdown-confluence#65] [Ideas]
Here is a list of thoughts for the future that might get supported.

Keen to hear which ones are important to people.

- [ ] Layout support (Confluence supports multiple columns) 
- [ ] Date macro support
- [ ] Decisions
- [ ] Status support
- [ ] Emoji
- [x] Add mention support
- [ ] Better smartlinks support (supporting the different types of smartlinks)
- [ ] Restricted permissions
- [ ] Render canvas to image
- [ ] Rendering emojis from Confluence in Obsidian
- [ ] Syncing pages from Confluence to Obsidian
- [ ] 2 way sync
- [ ] Convert tasklist to task list in Confluence
- [ ] Embed Page / Block - If you embed a page or block then copy content. Maybe there is a better option to use a macro to embed the content to link the pages together.
- [ ] Block links support - Linking to a block in another page being supported
- [ ] TOC Support

**Comment by jhilti** [comment]
Great work! support for on-premise instances would be great!

**Comment by wgregorian** [comment]
I'd love it if we could define variables that get populated on the build. For example, define a var in a JSON config (companyName = "Blah"), then use {{companyName}} in the markdown and populate it on publish.

**Comment by baaastijn** [comment]
Hello, first thanks for this plugin. Here is what's missing for me : support specific widgets from Material for MKdocs.

**Explanation :** 
In many projects, in many companies, the user documentation is (sadly) often started via Confluence. 
Then, some issues are raised (versioning, reuse, control over time, templating, etc) and we usually move the user documentation to markdown + static generator.

The preferred tool for this is MKdocs, from what is see after 10years (followed by docusaurus). 
And the preferred theme for MKDocs is https://squidfunk.github.io/mkdocs-material/

Quite often still, we need to mirror the markdown-doc to confluence, so that we can have a "unique search engine for everything"

**Where's the issue then ?** 
Material theme is widely used and very nice, but add some HTML parts for navigation for example, or emojis. 
They are improving the ergonomy, but not handled by this plugin

**Proposed idea 1**
Co-dev with MaterialDocs.

The maintainer of Material put in place a funding program. If you sponsor a bit the Material project your are granted specific features. I'm pretty sure that a feature called "Material doc to confluence" will be upvoted very quickly :) 

**Proposed idea 2**
Support Bootstrap Icon and grid system (div with col-*)

**Comment by darthwonka** [comment]
IDEA: Ability to override the `Confluence Parent Page ID` in the frontmatter to publish to confluence locations other than the default in the options. 
```
---
connie-publish: true
connie-parent-id: "035920532035"
connie-page-id: "3528947295729527"
---
```
